worker/periodicworker.go

```diff
@@ -68,8 +81,11 @@ func NewTimer(d time.Duration) PeriodicTimer {
 // sleeping for sleepDuration in between each call, until Kill() is called
 // The stopCh argument will be closed when the worker is killed. The error returned
 // by the doWork function will be returned by the worker's Wait function.
-func NewPeriodicWorker(call PeriodicWorkerCall, period time.Duration, timerFunc NewTimerFunc) worker.Worker {
+func NewPeriodicWorker(call PeriodicWorkerCall, period time.Duration, timerFunc NewTimerFunc, options ...PeriodicWorkerOption) worker.Worker {
 	w := &periodicWorker{newTimer: timerFunc}
+	for _, option := range options {
+		option(w)
+	}
```

Comment:
This is a little odd/cute/unexpected. You could have also replaced the `period` arg with a `func() time.Duration`. 

I'm not going to fight it too hard though.